base/stat.jl

```diff
@@ -61,13 +61,14 @@ show(io::IO, st::StatStruct) = print(io, "StatStruct(mode=$(oct(filemode(st),6))
 
 # stat & lstat functions
 
-const stat_buf = Array{UInt8}(ccall(:jl_sizeof_stat, Int32, ()))
+const stat_buf_sz = Int(ccall(:jl_sizeof_stat, Int32, ()))
```

Comment:
Use `Ref{NTuple{N,UInt8}}()` and the buffer and `Ref{NTuple{N,UInt8}}` as the `ccall` type.